core/storage/collection/gae_models.py

```diff
@@ -229,6 +229,28 @@ def _trusted_commit(
                     self.status == constants.ACTIVITY_STATUS_PRIVATE)
             ).put_async()
 
+    @classmethod
+    def export_data(cls, user_id):
+        """(Takeout) Export user-relevant properties of CollectionRightsModel.
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+                If the user_id is not valid, this method returns None.
+        Returns:
+            dict. The user-relevant properties of CollectionRightsModel in a
+                python dict format. In this case, we are returning all the ids
+                of collections that the user is connected to, so they either
+                own, edit, voice, or have permission to view.
+        """
+        collections = cls.get_all().filter(
+            ndb.OR(cls.owner_ids == user_id,
+                   cls.editor_ids == user_id,
+                   cls.voice_artist_ids == user_id,
+                   cls.viewer_ids == user_id)).fetch()
+        if not collections:  # Filter found no matching collections.
+            return None
+        collection_ids = [collection.key.id() for collection in collections]
+        return {'collection_ids': collection_ids}
```

Comment:
Shouldn't you separate by role? I.e. return a dict that's something like:
```
{
     owned_collection_ids: [...],
     editable_collection_ids: [...],
     ...
}
```

Ditto elsewhere.

Reply:
Ack